build error

I tried installation of your product.
But an error occurred by a build.



	git clone --recursive https://github.com/TaraHoleInIt/tarablessd1306_examples

	cd tarablessd1306_examples/

	cd HelloWorld/

	rm sdkconfig.*

	make menuconfig

	make

	Generating esp32.project.ld
	LD build/tarablessd1306_helloworld.elf
	/home/robotics/tarablessd1306_examples/HelloWorld/build/main/libmain.a(main.o):(.literal.DefaultBusInit+0x20): undefined reference to `SSD1306_I2CMasterInitDefault'
	/home/robotics/tarablessd1306_examples/HelloWorld/build/main/libmain.a(main.o):(.literal.DefaultBusInit+0x24): undefined reference to `SSD1306_I2CMasterAttachDisplayDefault'
	/home/robotics/tarablessd1306_examples/HelloWorld/build/main/libmain.a(main.o):(.literal.DefaultBusInit+0x28): undefined reference to `SSD1306_SPIMasterInitDefault'
	/home/robotics/tarablessd1306_examples/HelloWorld/build/main/libmain.a(main.o):(.literal.DefaultBusInit+0x2c): undefined reference to `SSD1306_SPIMasterAttachDisplayDefault'
	/home/robotics/tarablessd1306_examples/HelloWorld/build/main/libmain.a(main.o): In function `DefaultBusInit':
	/home/robotics/tarablessd1306_examples/HelloWorld/main/main.c:42: undefined reference to `SSD1306_I2CMasterInitDefault'
	/home/robotics/tarablessd1306_examples/HelloWorld/main/main.c:43: undefined reference to `SSD1306_I2CMasterAttachDisplayDefault'
	/home/robotics/tarablessd1306_examples/HelloWorld/main/main.c:43: undefined reference to `SSD1306_SPIMasterInitDefault'
	/home/robotics/tarablessd1306_examples/HelloWorld/main/main.c:48: undefined reference to `SSD1306_SPIMasterAttachDisplayDefault'
	collect2: error: ld returned 1 exit status
	/home/robotics/esp-idf/make/project.mk:517: recipe for target '/home/robotics/tarablessd1306_examples/HelloWorld/build/tarablessd1306_helloworld.elf' failed
	make: *** [/home/robotics/tarablessd1306_examples/HelloWorld/build/tarablessd1306_helloworld.elf] Error 1



There is default_if_i2c.c.

	$ grep -r SSD1306_I2CMasterInitDefault *
	components/tarablessd1306/ifaces/default_if_i2c.c:bool SSD1306_I2CMasterInitDefault( void ) {
	components/tarablessd1306/ssd1306_default_if.h:bool SSD1306_I2CMasterInitDefault( void );
	main/main.c:        assert( SSD1306_I2CMasterInitDefault( ) == true );


archive is:

	/tarablessd1306_examples/HelloWorld/build/tarablessd1306$ ar tv libtarablessd1306.a
	rw-rw-r-- 1000/1000  50236 Apr 18 11:25 2019 ssd1306_font.o
	rw-rw-r-- 1000/1000  55712 Apr 18 11:25 2019 ssd1306.o
	rw-rw-r-- 1000/1000  36044 Apr 18 11:25 2019 ssd1306_draw.o
	rw-rw-r-- 1000/1000  13624 Apr 18 11:25 2019 font_droid_sans_fallback_15x17.o
	rw-rw-r-- 1000/1000  12244 Apr 18 11:25 2019 font_droid_sans_mono_13x24.o
	rw-rw-r-- 1000/1000  18740 Apr 18 11:25 2019 font_liberation_mono_17x30.o
	rw-rw-r-- 1000/1000  17844 Apr 18 11:25 2019 font_droid_sans_mono_16x31.o
	rw-rw-r-- 1000/1000   8472 Apr 18 11:25 2019 font_droid_sans_fallback_11x13.o
	rw-rw-r-- 1000/1000   7532 Apr 18 11:25 2019 font_liberation_mono_9x15.o
	rw-rw-r-- 1000/1000  12244 Apr 18 11:25 2019 font_liberation_mono_13x21.o
	rw-rw-r-- 1000/1000  25048 Apr 18 11:25 2019 font_droid_sans_fallback_24x28.o
	rw-rw-r-- 1000/1000   6636 Apr 18 11:25 2019 font_droid_sans_mono_7x13.o
	rw-rw-r-- 1000/1000   2024 Apr 18 11:25 2019 default_if_spi.o
	rw-rw-r-- 1000/1000   2024 Apr 18 11:25 2019 default_if_i2c.o

 
Advice, please.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,3 +1,21 @@
-set(COMPONENT_SRCDIRS ". fonts ifaces")
-set(COMPONENT_ADD_INCLUDEDIRS ".")
-register_component()
+idf_component_register(
+  SRCS
+  "ssd1306.c"
+  "ssd1306_font.c"
+  "ssd1306_draw.c"
+  "ifaces/default_if_i2c.c"
+  "ifaces/default_if_spi.c"
+  "fonts/font_droid_sans_fallback_11x13.c"
+  "fonts/font_droid_sans_fallback_15x17.c"
+  "fonts/font_droid_sans_fallback_24x28.c"
+  "fonts/font_droid_sans_mono_13x24.c"
+  "fonts/font_droid_sans_mono_16x31.c"
+  "fonts/font_droid_sans_mono_7x13.c"
+  "fonts/font_liberation_mono_13x21.c"
+  "fonts/font_liberation_mono_17x30.c"
+  "fonts/font_liberation_mono_9x15.c"
+  "fonts/font_tarable7seg_16x32.c"
+  "fonts/font_tarable7seg_32x64.c"
+
+  INCLUDE_DIRS "."
+)

